drivers/sensor/ds18b20/ds18b20.c

```diff
@@ -0,0 +1,267 @@
+ */
+static int ds18b20_set_resolution(const struct device *dev, uint8_t resolution)
+{
+	struct ds18b20_data *data = dev->data;
+
+	data->scratchpad.config &= ~DS18B20_RESOLUTION_MASK;
+	data->scratchpad.config |= DS18B20_RESOLUTION(resolution);
+
+	return 0;
+}
+
+static int ds18b20_sample_fetch(const struct device *dev,
+				enum sensor_channel chan)
+{
+	const struct ds18b20_config *cfg = dev->config;
+	struct ds18b20_data *data = dev->data;
+	int status;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL ||
+			chan == SENSOR_CHAN_AMBIENT_TEMP);
+
+	if (!data->lazy_loaded) {
+		status = ds18b20_configure(dev);
+		if (status < 0) {
+			return status;
+		}
+		data->lazy_loaded = true;
+	}
+
+	status = ds18b20_temperature_convert(dev);
+	if (status < 0) {
+		LOG_DBG("W1 fetch error");
+		return status;
+	}
+	k_sleep(K_MSEC(measure_wait_ms[
+				DS18B20_RESOLUTION_INDEX(cfg->resolution)]));
+	ds18b20_read_scratchpad(dev, &data->scratchpad);
+
+	return 0;
+}
+
+static int ds18b20_channel_get(const struct device *dev,
+			       enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct ds18b20_data *data = dev->data;
+
+	if (chan == SENSOR_CHAN_AMBIENT_TEMP) {
+		ds18b20_temperature_from_raw((uint8_t *)&data->scratchpad.temp,
+					     val);
+	} else {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int ds18b20_configure(const struct device *dev)
+{
+	const struct ds18b20_config *cfg = dev->config;
+	struct ds18b20_data *data = dev->data;
+
+	/* In single drop configurations the rom can be read from device */
+	if (w1_get_peripheral_count(cfg->bus) == 1) {
+		if (w1_rom_to_uint64(&data->config.rom_id) == 0ULL) {
+			w1_read_rom(cfg->bus, &data->config.rom_id);
+		}
+	} else if (w1_rom_to_uint64(&data->config.rom_id) == 0ULL) {
+		LOG_DBG("nr: %d", w1_get_peripheral_count(cfg->bus));
+		LOG_ERR("ROM required, because multiple devices are on the bus");
+		return -EINVAL;
+	}
+
+	if ((cfg->family != 0) && (cfg->family != data->config.rom_id.family)) {
+		LOG_ERR("Found 1-Wire device is not a DS18B20");
+		return -EINVAL;
+	}
+
+	if (w1_reset_bus(cfg->bus) <= 0) {
+		LOG_ERR("No 1-Wire devices connected");
+		return -EINVAL;
+	}
+
+	ds18b20_read_power_supply(dev);
+	LOG_DBG("Using external power supply: %d", data->power_supply);
+
+	/* write defaults */
+	ds18b20_set_resolution(dev, cfg->resolution);
+	ds18b20_write_scratchpad(dev, data->scratchpad);
+	LOG_DBG("Init DS18B20: ROM=%016llx\n",
+		w1_rom_to_uint64(&data->config.rom_id));
+
+	return 0;
+}
+
+int ds18b20_attr_set(const struct device *dev,
+		     enum sensor_channel chan,
+		     enum sensor_attribute attr,
+		     const struct sensor_value *thr)
+{
+	struct ds18b20_data *data = dev->data;
+	int rc = 0;
+
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		rc = -ENOTSUP;
+	} else if ((enum sensor_attribute_w1)attr == SENSOR_ATTR_W1_ROM_ID) {
+		data->lazy_loaded = false;
+		w1_sensor_value_to_rom(thr, &data->config.rom_id);
+	};
+
+	return rc;
```

Comment:
I dont see the need of having `rc` here tbh, you could just return `-ENOTSUP` or `0` directly.